### ISSUE TITLE
False negative: TaintedCallable not fixed, when text checked against dynamically created array contents

### ISSUE THREAD

**Issue by tertek** [vimeo/psalm#9710]
Psalm Taint Analysis threw an error for TaintedCallable  for the following method `getMappedResultsBySource`. My fix by checking `$identifier` against `$allowedSources` only works for a hard coded array. When checking against a dynamically parsed array from JSON, the error remains, which I suppose is a false negative in my case, since `sources/sources.json` is safe to use.

```php
    private function getMappedResultsBySource($identifier, $data, $pid=null): ?array
    {

        //  Fix Tainted Call that could lead to running arbitrary functions ( https://psalm.dev/243 )
        
        //$allowedSources = self::getAllowedSources();    // psalm false negative
        $allowedSources = array_column((json_decode(file_get_contents("sources/sources.json")))->sources, "identifier");  // identical to getAllowedSources()
        //$allowedSources = ["my.source.1", "my.source.2"];    //hard coded works

        if(!in_array($identifier,  $allowedSources, true)) {
           return [];
        }
        ..

        //  Include class if not exists
        $path = $this->getSafePath(__DIR__ . "/sources/" . $class . ".source.php");
        if (!class_exists($class)) include_once($path);

        $class_with_namespace = "STPH\\addressAutoComplete\\" . $class;
        $arrayOfConstructorParameters = array($data, $format);

        return (new $class_with_namespace(...$arrayOfConstructorParameters))->getMappedResults();
    }

```

**Is there a way to tell psalm that my dynamically created array is fine? Or am I missing here something?**

**Comment by tertek** [comment]
> Hey @tertek, can you reproduce the issue on https://psalm.dev ?

Psalm runner output: 

Internal Psalm error on line 284 - /vendor/vimeo/psalm/src/Psalm/Internal/PhpVisitor/Reflector/ExpressionScanner.php: File includes are not allowed per your Psalm config - check the allowFileIncludes flag.
